### DISCUSSION TITLE
TabbedPage - Can't recreate Xamarin Android custom renderer.

### DISCUSSION THREAD

**Discussion by dustynlightfoot** [dotnet/maui#23820] [Upgrading from Xamarin]
Hey Folks,

I can't seem to recreate the custom bottom tab page experience I had in Xamarin, similar to what Steven Thewissen described here - https://thewissen.io/styling-the-android-bottom-tabs-animation-behavior-and-font.

When I try the TabbedViewHandler.Mapper.AppendToMapping route, I can only get a handle the Android ViewPager2 object, but there seems to be no way to get hold of the actual BottomNavigationView.

Not sure if I'm missing something or just on the wrong track 🤷‍♂️.

Oh and BTW if I just try and use my old custom renderer (as per the "Reuse custom renderers in .NET MAUI" documentation), I just get a very obscure "Specified cast is not valid" exception.

Kind Regards
Dustyn

**Comment by LeoJHarris** [comment]
If you need `BottomNavigationView` you can follow these links which might be helpful to you

https://github.com/dotnet/maui/discussions/16344](https://github.com/dotnet/maui/discussions/16344#discussioncomment-8294743)

https://stackoverflow.com/a/52833606
